### PR TITLE
docs: update storybook doc status, switchbutton and text

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Documentation

#### What is expected?

Rewrite documentation of status, switchButton and text to display more accurate code snippets and segment each stories in individual files.

### CODE REVIEW

File: src/components/Status/__stories__/Animated.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  variant: 'info',
+}
+Animated.parameters = {
+  docs: {
+    description: {
+      story:
+        'Animated prop will make StatusIndicator blink. This prop is usually used for temporary or "in progress" state.',
+    },
+  },
+}
```
Comment (baptisteArno) [inline src/components/Status/__stories__/Animated.stories.tsx:15]: Better to let it breath a bit don't you think?

```suggestion
export const Animated = Template.bind({})

Animated.args = {
  animated: true,
  variant: 'info',
}

Animated.parameters = {
  docs: {
    description: {
      story:
        'Animated prop will make StatusIndicator blink. This prop is usually used for temporary or "in progress" state.',
    },
  },
}
```

File: src/components/Text/__stories__/Variants.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { ComponentStory } from '@storybook/react'
+import Text, { textVariants } from '../index'
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const Variants: ComponentStory<typeof Text> = props => (
```
Comment (baptisteArno) [inline src/components/Text/__stories__/Variants.stories.tsx:5]: Pass props to `Text` and remove the eslint disable :)

**Review by baptisteArno — CHANGES_REQUESTED** [review]

File: src/components/Text/__stories__/Variants.stories.tsx (hunk 2 above)
Comment (matthprost) [reply src/components/Text/__stories__/Variants.stories.tsx:5]: @baptisteArno I cannot in this cas I will have errors on properties I passing to Text

Screenshot 2022-10-17 at 09 57 39

Comment (baptisteArno) [reply src/components/Text/__stories__/Variants.stories.tsx:5]: Does it work like this?

```
<Text as="div" key={variant} {...props} variant={variant}>
  {variant}
</Text>
```

Comment (matthprost) [reply src/components/Text/__stories__/Variants.stories.tsx:4]: @baptisteArno Yes it does!

**Review by baptisteArno — APPROVED** [review]

**Review by philibea — APPROVED** [review]

**Pull request merged**